filecoin-proofs/src/fr32_reader.rs

```diff
@@ -1,157 +1,105 @@
+use byte_slice_cast::*;
 use std::io;
 
-const DATA_BITS: u64 = 254;
-const TARGET_BITS: u64 = 256;
-
-#[derive(Debug)]
 pub struct Fr32Reader<R> {
     /// The source being padded.
     source: R,
-    /// How much of the target already was `read` from, in bits.
-    target_offset: u64,
-    /// Currently read byte.
-    buffer: Buffer,
+    /// Currently read block.
+    in_buffer: [u8; NUM_U128S_PER_BLOCK * 16],
```

Comment:
Instead of 16, you could explicitly calculate with `size_of` to show that the in and out buffers hold the same number of bytes.